releases_page/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io"
+	"net/http"
+	"os"
+)
+
+// Templates for main page and release page
+const (
+	mainPageTemplate = `<ul>
+	<li>
+		<a href="../">../</a>
+	</li>
+	{{ range . }}
+	<li>{{ $releaseName := .name | trimV }}
+		<a href="./{{ $releaseName }}/">{{ printf "tofu_%s" $releaseName }}</a>
+	</li>
+	{{ end }}
+	</ul>`
+	releasePageTemplate = `<ul>
+	<li>
+		<a href="../">../</a>
+	</li>
+	{{ range . }}
+	<li>
+		<a href="{{ .browser_download_url }}">{{ .name }}</a>
+	</li>
+	{{end}}
+	</ul>`
+)
+
+func main() {
+	os.Exit(generateReleasePages())
+}
+
+// Function to handle errors and close file
+func handleFileClose(file *os.File, err error) bool {
+	if err != nil {
+		if file != nil {
+			// Close the file if it's open
+			closeErr := file.Close()
+			if closeErr != nil {
+				fmt.Printf("Error closing file: %s\n", closeErr)
+			}
+		}
+		return true
+	}
+	return false
+}
+
+func generateReleasePages() int {
+	const (
+		githubAPIEndpoint = "https://api.github.com/repos/opentofu/opentofu/releases"
+		htmlFileName      = "index.html"
+	)
+
+	// Check for GitHub token in environment variable
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		fmt.Println("Failed to retrieve Authorization token from environment variable. Please ensure that 'GITHUB_TOKEN' is set properly.")
+		return 1
+	}
```

Comment:
Tiny tiny nitpick: maybe if the token is not set, we could just not set the `Authorization` header below? That way it would be easier to test (but may run into a rate limit).

Reply:
Yes, GitHub allows it too, but may run into errors with rate limits.

Reply:
implemented changes in [8d7a660](https://github.com/opentofu/get.opentofu.org/pull/22/commits/8d7a660122b272f0493f711415607fe87b292adb)